src/js/node.rs

```diff
@@ -0,0 +1,220 @@
+use std::{collections::HashMap, path::PathBuf};
+
+use crate::{
+    ConnType, ConnectionTypeMixed, CounterStats, DirectAddrInfo, IrohNode, NodeStatusResponse,
+    PublicKey,
+};
+
+use futures::TryStreamExt;
+use napi_derive::napi;
+
+#[napi]
+impl DirectAddrInfo {
+    /// Get the reported latency, if it exists, in milliseconds
+    #[napi(js_name = "latency")]
+    pub fn latency_js(&self) -> Option<u32> {
+        self.0.latency.map(|l| l.as_millis() as _)
+    }
+
+    /// Get the last control message received by this node
+    #[napi(js_name = "lastControl")]
+    pub fn last_control_js(&self) -> Option<JsLatencyAndControlMsg> {
+        self.0
+            .last_control
+            .map(|(latency, control_msg)| JsLatencyAndControlMsg {
+                latency: latency.as_millis() as _,
+                control_msg: control_msg.to_string(),
+            })
+    }
+
+    /// Get how long ago the last payload message was received for this node in milliseconds.
+    #[napi(js_name = "lastPayload")]
+    pub fn last_payload_js(&self) -> Option<u32> {
+        self.0.last_payload.map(|d| d.as_millis() as _)
+    }
+}
+
+/// The latency and type of the control message
+#[napi(constructor, js_name = "LatencyAndControlMsg")]
+pub struct JsLatencyAndControlMsg {
```

Comment:
we can drop the `Js` prefix given the scope I believe

Reply:
so removing the `Js` makes cargo unhappy unless we mark all of the associated structs in the main lib with `feature = "no-napi"`

which we can do, but I'd rather leave it as is if that's okay with you.